include/circt/Dialect/RTL/RTLTypes.td

```diff
@@ -86,3 +86,66 @@ class InOutElementConstraint<string value, string inoutValue>
   : TypesMatchWith<"type should be element of inout type",
                    inoutValue, value,
                    "$_self.cast<InOutType>().getElementType()">;
+
+// Adapted from TestTypeDefs.td
+// A parent type for any type which is just a list of fields (e.g. structs,
+// unions).
+def StructType : RTLType<"Struct"> {
+  let summary = "RTL struct type";
+  let description = [{
+    Struct type has value semantics, updates to the struct are logically by 
+    copy.
```

Comment:
Well this is up to the choice of ops that we support on them.  In any case, please describe what a struct *is*: it's an ordered value semantic collection of fields.